README.md

```diff
@@ -4,45 +4,120 @@ The goal is to improve payment confirmation experience with the help of WebAuthn
 
 ![Screenshot](https://raw.githubusercontent.com/rsolomakhin/secure-payment-confirmation/master/payment.png)
 
+The rest of the document is organized into these sections:
+
+- [Creating a credential](#creating-a-credential)
+- [Querying a credential](#querying-a-credential)
+- [Authenticating a payment](#authentication-a-payment)
+
 ## Creating a credential
 
-A new credential type is introduced for `navigator.credentials.create()` that stores the name and the icons of a payment instrument.
+A new `SecurePaymentCredential` credential type is introduced for `navigator.credentials.create()` to bind descriptions of a payment instrument, i.e. a name and an icon, with a vanilla [PublicKeyCredential].
+
+Proposed new spec that extends [Web Authentication]:
+```webidl
+[SecureContext, Exposed=Window]
+interface SecurePaymentCredential : PublicKeyCredential {
+};
 
+partial dictionary CredentialCreationOptions {
+  SecurePaymentCredentialCreationOptions securePayment;
+};
+
+dictionary SecurePaymentCredentialCreationOptions {
+  required PublicKeyCredentialRpEntity rp;
+  required SecurePaymentCredentialInstrument instrument;
+  required BufferSource challenge;
+  required sequece<PublicKeyCredentialParameters> pubKeyCredParams;
+  unsigned long timeout;
+  
+  // PublicKeyCredentialCreationOption attributes that are intentionall omitted:
+  // user: For a SecurePaymentCredential, |instrument| is analogous to |user|.
+  // excludeCredentials: No payment use case has been proposed for this field.
+  // attestation: Authenticator attestation is considered an anti-pattern for adoption so will not be supported.
+  // extensions: No payment use case has been proposed for this field.
+};
+
+dictionary SecurePaymentCredentialInstrument {
+  required DOMString instrumentId;
+  required DOMString displayName;
+  required USVString icon;
+};
+```
+
+Example usage:
 ```javascript
-const publicKeyCredentialCreationOptions = {
-  paymentInstrument: {
-    name: 'Mastercard****4444',
-    icons: [{
-  	  'src': 'icon.png',
-  	  'sizes': '48x48',
-  	  'type': 'image/png',
-    }],
+const securePaymentConfirmationCredentialCreationOptions = {
+  instrument: {
+    instrumentId: "Q1J4AwSWD4Dx6q1DTo0MB21XDAV76",
+    displayName: 'Mastercard****4444',
```

Comment:
I'm worried about these kinds of display names... they are somewhat easy to screw up (and using "****" is a bit ugly). Figuring out how to obfuscate the display of a payment credential is something we need to figure out.

Reply:
the "displayName" might need to be structured somehow... we might need to look at all sorts of instruments to figure out common bits of information.

Reply:
Unicode has support for better looking characters, such as the [middle dot](https://en.wikipedia.org/wiki/Interpunct), which is used in Chrome UI to obfuscate credit cards, the format of which is well known. Callers of this API can use the middle dot as well, especially if we put it in public documentation.

My hope is this API will be used for more than only credit cards, so the best obfuscation is hard to guess ahead of time.

Reply:
> My hope is this API will be used for more than only credit cards, so the best obfuscation is hard to guess ahead of time.

+1. The caller of this API is an issuer and they are incentivized to figure out the best way to obfuscate the payment instrument ID and with unicode, has the tools to generate good looking strings. IMHO, imposing more structure here has the cost of less flexibility and tighter coupling between browser and the payment instrument provider, with very small UX gain.